### PR TITLE
feat(DTFS2-6307): Bond Party URN tests

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction
Bond issuer and beneficiary both were missing various E2E imperative assertions alongside with improper inline error.

## Resolution
* Enhanced E2E coverage.
* Resolved inline error reference.
* Updated unit tests.

### CODE REVIEW

File: e2e-tests/trade-finance-manager/cypress/e2e/journeys/case/parties/bond-beneficiary.spec.js [hunk 1]
```diff
@@ -38,14 +38,178 @@ context('Bond beneficiary URN - User can add, edit, confirm and submit URN to th
...
+
+        pages.bondBeneficiaryPage.urnInput(1).clear().type(' ');
+        pages.bondBeneficiaryPage.urnInput(2).clear().type(' ');
+        pages.bondBeneficiaryPage.saveButton().click();
+
+        cy.url().should('eq', relative(`/case/${dealId}/parties/${party}`));
+        pages.bondBeneficiaryPage.errorSummary().contains('Enter a minimum of 3 numbers');
+        pages.bondBeneficiaryPage.urnError(1).contains('Enter a minimum of 3 numbers');
+        pages.bondBeneficiaryPage.urnError(2).contains('Enter a minimum of 3 numbers');
+      });
```
Comment (ttbarnes) [inline e2e-tests/trade-finance-manager/cypress/e2e/journeys/case/parties/bond-beneficiary.spec.js:143]: Not a blocker - would be nice to split these up into their own "it's" for easier debugging; Could also create a new command to save repetition e.g `submitAndAssertValidationError(field, message)`

File: trade-finance-manager-ui/server/controllers/case/parties/helpers.js [hunk 2]
```diff
@@ -11,30 +11,20 @@ const userCanEdit = (user) => userIsInTeam(user, [CONSTANTS.TEAMS.BUSINESS_SUPPO
 const bondType = (party) => (party === 'bond-issuer' ? 'bondIssuerPartyUrn' : 'bondBeneficiaryPartyUrn');
 
 // checks if bond type and returns true or false
-const isBondPartyType = (partyType) => (partyType === 'bondBeneficiaryPartyUrn' || partyType === 'bondIssuerPartyUrn');
+const isBondPartyType = (partyType) => (partyType === 'bond-beneficiary' || partyType === 'bond-issuer');
```
Comment (ttbarnes) [inline trade-finance-manager-ui/server/controllers/case/parties/helpers.js:14]: constant?

File: trade-finance-manager-ui/server/controllers/case/parties/helpers.js [hunk 3]
```diff
@@ -11,30 +11,20 @@ const userCanEdit = (user) => userIsInTeam(user, [CONSTANTS.TEAMS.BUSINESS_SUPPO
...
+const constructErrRef = (party, index) => (isBondPartyType(party) ? `partyUrn-${index}` : 'partyUrn');
 
 // checks if string is empty - checks that string is '' and does not have a length and returns true if so
-const isEmptyString = (str) => {
-  if (!str || ((typeof str === 'string' || str instanceof String) && !str.trim().length)) {
-    return true;
-  }
-  return false;
-};
+const isEmptyString = (str) => (!str || ((typeof str === 'string' || str instanceof String) && !str.trim().length) ? true : false);
```
Comment (ttbarnes) [inline trade-finance-manager-ui/server/controllers/case/parties/helpers.js:27]: Could be moved into generic helper directory

File: e2e-tests/trade-finance-manager/cypress/e2e/journeys/case/parties/bond-issuer.spec.js [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
+import { T1_USER_1, BUSINESS_SUPPORT_USER_1 } from '../../../../../../e2e-fixtures';
 import { MOCK_MAKER_TFM, ADMIN_LOGIN } from '../../../../fixtures/users-portal';
 
 context('Bond issuer URN - User can add, edit, confirm and submit URN to the TFM', () => {
   let dealId;
   const dealFacilities = [];
   const party = 'bond-issuer';
-  // const mockUrn = ['1234', '1234'];
-  // const partyUrn = ['00307249', '00307249'];
+  const mockUrn = ['1234', '1234'];
```
Comment (Zainzzkk) [inline e2e-tests/trade-finance-manager/cypress/e2e/journeys/case/parties/bond-issuer.spec.js:11]: can we put these in constants?

File: trade-finance-manager-ui/server/controllers/case/parties/helpers.js (hunk 3 above)
Comment (abhi-markan) [reply trade-finance-manager-ui/server/controllers/case/parties/helpers.js:27]: Good shout, will tackle this in a separate PR a tech-debt ticket already exist regarding same.

**Review by Zainzzkk — APPROVED** [review]
LGTM

**Pull request merged**